### ISSUE TITLE
Configurable `x_for` for both the WSGI app and for botdetection

### ISSUE THREAD

**Issue by rwjack** [searxng/searxng#3632]
Currently, only botdetection has a configurable `x_for` in `limiter.toml`.

In my use case of `client->proxy1->proxy2->searx`, I have to configure `x_for = 2`, for the limiter to work, which then creates an endless log spam of:
```
2024-07-02 14:58:00,377 WARNING:searx.botdetection: IP from WSGI environment (proxy2) is not equal to IP from X-Real-IP (client)
2024-07-02 14:58:02,093 WARNING:searx.botdetection: IP from WSGI environment (proxy2) is not equal to IP from X-Forwarded-For (client)
```
Warning found here: https://github.com/searxng/searxng/blob/39aaac40d63d53555b7fc12d56f0825c85ecf567/searx/botdetection/_helpers.py#L123-L124

---

As per @return42's recommendation:

> SearXNG uses the [ProxyFix](https://werkzeug.palletsprojects.com/en/3.0.x/middleware/proxy_fix/) with it default x_for=1 .. its implemented here in this line: https://github.com/searxng/searxng/blob/39aaac40d63d53555b7fc12d56f0825c85ecf567/searx/flaskfix.py#L77

Meaning the ideal solution would be to make the `x_for` parameter configurable in `settings.yml` for Searx, instead of `limiter.toml`, which would set the parameter for both the UWSGI app and for the limiter.

**Comment by nowheretobefound** [comment]
I have the same issue with the log spam after setting limiter.toml x_for=2

```
WARNING:searx.botdetection: IP from WSGI environment (proxy2) is not equal to IP from X-Forwarded-For (client)
```

But I didnt understand  @return42's recommendation: Is there a workaround for this issue ( app.wsgi_app = ReverseProxyPathFix(ProxyFix(app.wsgi_app))  with default x_for=1) or  do we have to wait for a fix to get x_for=2 for the wsgi_app ?

**Comment by return42** [comment]
> or do we have to wait for a fix to get x_for=2 for the wsgi_app 

Yes, sadly this is not implemented yet .. and this issue here is the request to implement it.

**Comment by return42** [comment]
closed by https://github.com/searxng/searxng/pull/4911

@nowheretobefound @rwjack please have a look at the PR .. the `x_for` option is no longer needed/supported. If needed, you can add your proxies to the `botdetection.trusted_proxies` list in the [limiter.toml](https://docs.searxng.org/admin/searx.limiter.html#limiter-toml)

**Comment by nowheretobefound** [comment]
Thank you,  I updated limiter.tom with trusted_proxies and removed real_ip and x_for, and nothing in the logs and the ip search returns my correct ip, so all good it seems for my extremely limited testing :)
